packages/fetchai/skills/echo/handlers.py

```diff
@@ -53,3 +64,76 @@ def teardown(self) -> None:
         :return: None
         """
         self.context.logger.info("Echo Handler: teardown method called.")
+
+    def _handle_unidentified_dialogue(self, message: DefaultMessage) -> None:
+        """
+        Handle unidentified dialogue.
+
+        :param message: the message.
+        :return: None
+        """
+        self.context.logger.info(
+            "received invalid default message={}, unidentified dialogue.".format(
+                message
+            )
+        )
+        dialogues = cast(DefaultDialogues, self.context.default_dialogues)
+        reply = DefaultMessage(
+            performative=DefaultMessage.Performative.ERROR,
+            dialogue_reference=dialogues.new_self_initiated_dialogue_reference(),
+            error_code=DefaultMessage.ErrorCode.INVALID_DIALOGUE,
+            error_msg="Invalid dialogue.",
+            error_data={"default_message": message.encode()},
+        )
+        reply.to = message.sender
+        reply.counterparty = message.sender
+        dialogues.update(reply)
+        self.context.outbox.put_message(message=reply)
+
+    def _handle_error(self, message: DefaultMessage, dialogue: DefaultDialogue) -> None:
+        """
+        Handle a message of error performative.
+
+        :param message: the default message.
+        :param dialogue: the dialogue.
+        """
+        self.context.logger.info(
+            "received default error message={} in dialogue={}.".format(
+                message, dialogue
+            )
+        )
+
+    def _handle_bytes(self, message: DefaultMessage, dialogue: DefaultDialogue):
+        """
+        Handle a message of bytes performative.
+
+        :param message: the default message.
+        :param dialogue: the default dialogue.
+        :return: None
+        """
+        self.context.logger.info(
+            "Echo Handler: message={}, sender={}".format(message, message.counterparty)
+        )
+        reply = DefaultMessage(
+            performative=DefaultMessage.Performative.BYTES,
+            dialogue_reference=message.dialogue_reference,
```

Comment:
it's better to do `dialogue.dialogue_label.dialogue_reference` as then you get the full reference if available